### PR TITLE
Improve clarity of dev environment setup doc

### PR DESCRIPTION
**Pull request by keval6b**
# Details
Inspired by #3040
The doc is not clear on what is required for a local dev environment. Hopefully this is an improvement

## Changes
Rewrites one page in the documentation

### CODE REVIEW

File: website/developer-docs/setup/full-dev-environment.md [hunk 1]
```diff
@@ -46,18 +41,23 @@ Before committing code, run `make lint` to ensure your code is formatted well. T
...
 ```
 cd web/
 npm i
 npm run build
 ```
 
-If you want to make changes to the UI, run `npm run watch` instead.
+If you want to live-edit the UI, you should run `make web-watch` instead. This will immediately update the UI with any changes you make so you can see the results in real time without needing to rebuild.
+
+To format the frontend code, run `npx eslint . --fix` and `npm run lit-analyse`.
```
Comment (BeryJu) [inline website/developer-docs/setup/full-dev-environment.md:58]: This can actually be replaced by 
```
run `make web`
```

**Review by BeryJu — COMMENTED** [review]
Thanks for helping to simplify this

**Comment by thechubbypanda** [comment]
Hang on, does `npm run extract` build the UI files?

What's the difference between `npm run build` and `make web-install`?

**Comment by BeryJu** [comment]
npm run extract extracts the language strings for translation 

make web-install installs the web dependencies (there should probably be a note about running make install instead of running poetry install)

**Comment by thechubbypanda** [comment]
Ok how's this now?

File: Makefile [hunk 2]
```diff
@@ -103,6 +103,9 @@ run:
 ## Web
 #########################
 
+web-build:
```
Comment (BeryJu) [inline Makefile:106]: I'd change it to

```
web-build: web-install
    cd web && npm run build
```

File: website/developer-docs/setup/full-dev-environment.md [hunk 3]
```diff
@@ -2,21 +2,21 @@
...
-
-### Requirements
+## Requirements
 
 -   Python 3.10
 -   poetry, which is used to manage dependencies, and can be installed with `pip install poetry`
 -   Go 1.18
 -   PostgreSQL (any recent version will do)
 -   Redis (any recent version will do)
+-   Node 14 (or later)
```
Comment (BeryJu) [inline website/developer-docs/setup/full-dev-environment.md:12]: Also probably not a big different, but while we're at it, authentik now uses node 16

**Comment by BeryJu** [comment]
Oh, the indent in the Makefile needs to be with tabs not spaces, hence the CI jobs are failing

Also you might want to run `make website` (which I need to add to the documentation guide) to format the website markdown

**Pull request merged**